api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+
+	// PlatformConfigured indicates (if status is true) that the
+	// platform configuration specified for the platform provider has been applied
+	PlatformConfigured ConditionType = "PlatformInfrastructureConfigured"
+
+	// PlatformIAMConfigured indicates (if status is true) that the IAM is configured
+	PlatformIAMConfigured ConditionType = "PlatformIAMConfigured"
+)
+
+func init() {
+	SchemeBuilder.Register(&HostedClusterInfrastructure{}, &HostedClusterInfrastructureList{})
+}
+
+// HostedClusterInfrastructureSpec is the desired behavior of a HostedClusterInfrastructure.
+type HostedClusterInfrastructureSpec struct {
+
+	// InfraID is a globally unique identifier for the cluster. This identifier
+	// will be used to associate various cloud resources with the HostedCluster
+	// and its associated NodePools. If not specified the metadata.name for this
+	// resource is used. When specified, this value is used.
+	//
+	// +optional
+	// +immutable
+	InfraID string `json:"infraID,omitempty"`
+
+	// SubDomain is the identifier that is used between the *.apps and base-domain
+	// when building an ingress URL. This will also be the name of the cluster.
+	// If no value is provided, the infra-id is used
+	// When SubDomain = my and base-domain = company.com, the resulting ingress URL
+	// will be *.apps.my.company.com
+	//
+	// +optional
+	// +immutable
+	SubDomain string `json:"subDomain,omitempty"`
+
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform PlatformInfraSpec `json:"platform"`
+
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
+	BaseDomain string `json:"baseDomain"`
+
+	// CloudProvider secret, contains the Cloud credenetial and Base Domain
+	// When not present, we expect all values to populated at create time
+	// This can be from the hypershift cli or via a kubectl create.
+	// +optional
+	CloudProvider corev1.LocalObjectReference `json:"cloudProvider,omitempty"`
+}
+
+// PlatformSpec specifies the underlying infrastructure provider for the cluster
+// and is used to configure platform specific behavior.
+type PlatformInfraSpec struct {
+	// Type is the type of infrastructure provider for the cluster.
+	//
+	// +kubebuilder:validation:Enum=AWS;Azure;PowerVS
+	// +immutable
+	Type PlatformType `json:"type"`
+
+	// AWS specifies configuration for clusters running on Amazon Web Services.
+	//
+	// +optional
+	// +immutable
+	AWS *AWSPlatformInfraSpec `json:"aws,omitempty"`
+
+	// Azure defines azure specific settings
+	Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+	// PowerVS specifies configuration for clusters running on IBMCloud Power VS Service.
```

Comment:
does it exist a use case for managing infra this way in powerVS?

Reply:
That is my understanding. There is a create infra powervs. I was having discussions with them, but I haven't dived into it yet.